### ISSUE TITLE
Action output not helpful

### ISSUE THREAD

**Issue by andig** [golangci/golangci-lint-action#965]
### Welcome

- [X] Yes, I understand that the GitHub action repository is not the repository of golangci-lint itself.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc).

### Description of the problem

Lint action output is not helpful in that
- issues are not visible at first sight when looking into the action log and
- issues apparently sometimes omit filenames

<img width="381" alt="Screenshot 2024-02-08 at 15 27 17" src="https://github.com/golangci/golangci-lint-action/assets/184815/ab673ac6-cdcf-4cb4-97c4-ec6d439b45b8">

### Version of golangci-lint

Not sure

### Version of the GitHub Action

golangci/golangci-lint-action@v3

### Workflow file

<details>

https://github.com/evcc-io/evcc/blob/0631f600926faa0fc3566c3ac9ff2c6c8fa8c8ed/.github/workflows/default.yml#L98-L102

https://github.com/evcc-io/evcc/blob/0631f600926faa0fc3566c3ac9ff2c6c8fa8c8ed/.golangci.yml

</details>

### Go version

1.22.0

### Code example or link to a public repository

<details>

https://github.com/evcc-io/evcc/actions/runs/7830801405/job/21365710661?pr=11007

</details>

**Comment by gaby** [comment]
I'm having the same problem, there's no output. Enabling annotation makes 0 difference.

**Comment by iamnande** [comment]
@andig this issue started this week for me as well. although the `golangci-lint` run itself passes; overall the job still fails.

![Screenshot 2024-02-09 at 13 10 15](https://github.com/golangci/golangci-lint-action/assets/7806510/52975994-a77c-41c2-97fe-60c580829696)

**Comment by ldez** [comment]
Since v3.6.0 of the action, you can now provide extra args to add extra output format.

Example:

```yaml
    - name: Golangci Lint
      uses: golangci/golangci-lint-action@v4
      with:
        version: v1.56.2
        args: --out-format=colored-line-number
```

https://golangci-lint.run/usage/configuration/#output-configuration

Related to #769

Duplicate of #618, #119

**Comment by andig** [comment]
@ldez thank you. It would still be nice to see the linter messages during the run phase right in the log without needing to expand the "run golangci-lint" section.

**Comment by ldez** [comment]
The problem will be fixed with golangci-lint v1.58: the file information (path, position) will be displayed.

https://github.com/golangci/golangci-lint/pull/4685

**Comment by ldez** [comment]
I added a new option `annotations`, which may be useful to some of you.

I added this option for several reasons:
- The number of annotations is [limited](https://github.com/actions/toolkit/blob/main/docs/problem-matchers.md#limitations).
- Some people don't want to use annotations and just want the classic `colored-line-number` format.

If you set `annotations` to `false`, the output format(s) will follow the golangci-lint configuration file (or CLI flags from `args`) and use the same default as golangci-lint (i.e. `colored-line-number`).

https://github.com/golangci/golangci-lint-action?tab=readme-ov-file#annotations
